Automattic/woocommerce-payments
# Add helper function/method for raw amount conversion

Fixes #5524 

#### Changes proposed in this Pull Request

This PR adds the `get_raw_conversion` method that allows for conversion of an amount from one currency to another without any sort of rounding or price charm added. This helps with compatibility issues related to third party developer plugins.

#### Testing instructions

* The tests assume that your store's currency is USD.
* Navigate to WooCommerce > Settings > Multi-Currency and then enable CAD and EUR if not enabled.
  * CAD's conversion rate should be the local default: 1 USD = 1.288345 CAD
  * EUR's conversion rate should be the local default: 1 USD = 0.913827 EUR
* Add the below snippet to your site.
* After the snippet is added, any page on the front end.
* Check your `debug.log` to find output similar to the below.
  * CAD $amount: 12.88345
  * EUR CAD $amount: 7.0930302054186

Snippet:
```
add_action( 'wp_footer', function() {
	if ( ! function_exists( 'WC_Payments_Multi_Currency' ) ) {
		return;
	}
	$amount = WC_Payments_Multi_Currency()->get_raw_conversion( 10.00, 'CAD' );
	error_log( 'CAD $amount: ' . print_r( $amount, true ) );
	$amount = WC_Payments_Multi_Currency()->get_raw_conversion( 10.00, 'EUR', 'CAD' );
	error_log( 'EUR CAD $amount: ' . print_r( $amount, true ) );
});
```
-------------------
